controller/setting_controller.go

```diff
@@ -1068,3 +1079,367 @@ func (sc *SettingController) cleanupFailedSupportBundles() error {
 
 	return nil
 }
+
+func (sc *SettingController) GetCheckUpgradeRequestExtraInfo() (CheckUpgradeRequestExtraInfo, error) {
+	extraInfo := &ExtraInfo{
+		logger:        sc.logger,
+		ds:            sc.ds,
+		kubeClient:    sc.kubeClient,
+		metricsClient: sc.metricsClient,
+		structFields:  util.StructFields{},
+		controllerID:  sc.controllerID,
+		namespace:     sc.namespace,
+	}
+
+	kubeVersion, err := sc.kubeClient.Discovery().ServerVersion()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get Kubernetes server version")
+	}
+	extraInfo.structFields.Append(ExtraInfoKubernetesVersion, kubeVersion.GitVersion, extraInfo.structFields.ConvertTag(ExtraInfoKubernetesVersion))
```

Comment:
NIT: Should not this be a part of `ClusterScope` info?

Reply:
While it could be included in the ClusterScope, I've separated it from other usage information so it won't be excluded by the `Allow Collecting Longhorn Usage Metrics` setting, considering we are already collecting the Kubernetes versions.